iDiff/utils/package_diff_utils.go

```diff
@@ -1,11 +1,31 @@
 package utils
 
 import (
+	"fmt"
 	"io/ioutil"
 	"path/filepath"
 	"reflect"
+	"strings"
+
+	"github.com/golang/glog"
 )
 
+// MultiVersionPackageDiff stores the difference information between two images which could have multi-version packages.
+type MultiVersionPackageDiff struct {
+	Image1    string
+	Packages1 map[string]map[string]PackageInfo
+	Image2    string
+	Packages2 map[string]map[string]PackageInfo
+	InfoDiff  []MultiVersionInfo
+}
+
+// MultiVersionInfo stores the information for one multi-version package in two different images.
```

Comment:
just to clarify-- so is it true that multi-version packages would exist across images, not within? because in your tests, it looks like there's a multi-version package example within one image...